lib/Property.php

```diff
@@ -319,6 +319,77 @@ function jsonSerialize() {
         );
     }
 
+    /**
+     * Returns the value, in the format it should be encoded for XML.
+     *
+     * This method must always return an array.
+     *
+     * @return array
+     */
+    function getXmlValue() {
+
+        return $this->getJsonValue();
+
+    }
+
+    /**
+     * Sets the XML value, as it would appear in a xCard or xCal object.
+     *
+     * The value must always be an array.
+     *
+     * @param array $value
+     * @return void
+     */
+    function setXmlValue(array $value) {
```

Comment:
All `getXmlValue` methods have been removed.
All `xmlSerialize` methods now receive a `Sabre\Xml\Writer` object.
The `setXmlValue` are still based on array because the XML parser works with an array. These is 1 definition of `setXmlValue` and only 4 overwrites. I think it's fine enough.